roles/mongodb/tasks/mongodb_auth.yml

```diff
@@ -0,0 +1,68 @@
+---
+- name: Ensure pymongo is installed for the mongodb_user ansible module
+  become: yes
+  pip:
+    name: pymongo
+
+- name: See if authorization is enabled in mongod.conf
+  check_mode: yes
+  lineinfile:
+    path: /etc/mongod.conf
+    insertafter: 'security:'
+    line: '  authorization: enabled'
+  register: _mongo_authorization
+  # changed = line not in file, authorization is disabled
+  # succeeded = line in file, authorization is enabled
+
+- name: Add mongo admin
+  mongodb_user:
+    state: present
+    update_password: on_create
+
+    name: "{{ mongodb_admin_username }}"
+    password: "{{ mongodb_admin_password }}"
+    database: "{{ mongodb_admin_db }}"
+    roles: userAdminAnyDatabase
+
+    login_host: "{{ mongodb_host }}"
+    login_port: "{{ mongodb_port }}"
+    login_user: "{{ _mongo_authorization|changed | ternary(omit, mongodb_admin_username) }}"
+    login_password: "{{ _mongo_authorization|changed | ternary(omit, mongodb_admin_password) }}"
+
+- name: Enable security section in mongod.conf
+  become: yes
+  replace:
+    path: /etc/mongod.conf
+    regexp: '^#security:'
+    replace: 'security:'
+
+- name: Enable authentication in mongod.conf
+  become: yes
+  lineinfile:
+    path: /etc/mongod.conf
+    insertafter: '^security:'
+    line: '  authorization: enabled'
+  notify: restart mongod
```

Comment:
The correct handler name is: `restart mongodb`, according to https://github.com/StackStorm/ansible-st2/blob/master/roles/mongodb/handlers/main.yml